python/lsst/cp/pipe/defects.py

```diff
@@ -529,6 +539,48 @@ def findHotAndColdPixels(self, exp, imageType, setMask=False):
             footprintList += mergedSet.getFootprints()
 
         defects = measAlg.Defects.fromFootprintList(footprintList)
+
+        # If there's a column with on and off defects, mask all the pixels in between.
+        # Get the x, y values of the defects.
+        x, y = [], []
+        for defect in defects:
+            bbox = defect.getBBox()
+            x.append(bbox.getMinX())
+            y.append(bbox.getMinY())
+        x = np.array(x)
+        y = np.array(y)
+
+        # Find the defects with same "x" (vertical) coordinate (column).
+        unique, counts = np.unique(x, return_counts=True)
+        colThreshold = self.config.colThreshold
+        gapThreshold = self.config.gapThreshold  # (not "bad" pixels, see below).
+        multipleX = []
+        for (a, b) in zip(unique, counts):
+            if b > colThreshold:
+                multipleX.append(a)
+        if len(multipleX) != 0:
+            for x0 in multipleX:
+                index = np.where(x == x0)
+                multipleY = y[index]  # multipleY and multipleX are in 1-1 correspondence.
+                minY, maxY = np.min(multipleY), np.max(multipleY)
+
+                # Next few lines: don't mask pixels in column if gap between
+                # two consecutive bad pixels is too large.
+                diffIndex = np.where(np.diff(multipleY) > gapThreshold)[0]
+                if len(diffIndex) != 0:
+                    limits = [minY]  # put the minimum first
+                    for gapIndex in diffIndex:
+                        limits.append(multipleY[gapIndex])
+                        limits.append(multipleY[gapIndex+1])
+                    limits.append(maxY)  # maximum last
```

Comment:
Could you do the above loop with a `sort(multipleY[diffIndex])` instead? I'm having trouble visualizing it.

It also seems like you could create the defects themselves as part of this loop, and get rid of the `for i in np.arange` below.

Reply:
In the end I left the code as it was, because I thought it was easy to understand as it is. I have added clarifications in the comments and doc-string; please let me know if further clarification is needed.